src/tests/ftest/cart/util/cart_utils.py

```diff
@@ -418,6 +418,50 @@ def build_cmd(self, env, host, **kwargs):
 
         return tst_cmd
 
+    @staticmethod
+    def log_check_valgrind_memcheck(self):
+        """Check valgrind memcheck log files for errors."""
+
+        logparse = self.params.get("logparse", "/run/tests/*/")
+
+        memcheck_errors = 0
+
+        strict_test = False
+        self.log.info("Parsing log path %s", self.log_path)
+        if not os.path.exists(self.log_path):
+            self.log.info("Path does not exist")
+            return
+
+        xml_filename_fmt = r"^valgrind\.\S+\.memcheck$"
+        memcheck_files = list(filter(lambda x: re.match(xml_filename_fmt, x),
+                                os.listdir(self.log_path)))
+
+        for filename in memcheck_files:
+
+            log_file = os.path.join(self.log_path, filename)
+
+            file1 = open(log_file, 'r')
```

Comment:
(pylint-consider-using-with) Consider using 'with' for resource-allocating operations